lib/install/webpacker.rb

```diff
@@ -0,0 +1,128 @@
+# verify that all critical dependencies are up to date; if not, queue for later
+package_list = Rails.root.join("tmp/stimulus_reflex_installer/npm_package_list")
+dev_package_list = Rails.root.join("tmp/stimulus_reflex_installer/npm_dev_package_list")
+package_json = Rails.root.join("package.json")
+lines = File.readlines(package_json)
+
+if !lines.index { |line| line =~ /^\s*["']webpack["']: ["']\^4.46.0["']/ }
+  FileUtils.touch(package_list)
+  append_file(package_list, "webpack@^4.46.0\n", verbose: false)
+  say "✅ Enqueued webpack@^4.46.0 to be added to dependencies"
+end
+
+if !lines.index { |line| line =~ /^\s*["']webpack-cli["']: ["']\^3.3.12["']/ }
+  FileUtils.touch(package_list)
+  append_file(package_list, "webpack-cli@^3.3.12\n", verbose: false)
+  say "✅ Enqueued webpack-cli@^3.3.12 to be added to dependencies"
+end
+
+if !lines.index { |line| line =~ /^\s*["']@rails\/webpacker["']: ["']\^5.4.3["']/ }
+  FileUtils.touch(package_list)
+  append_file(package_list, "@rails/webpacker@^5.4.3\n", verbose: false)
+  say "✅ Enqueued @rails/webpacker@^5.4.3 to be added to dependencies"
+end
+
+if !lines.index { |line| line =~ /^\s*["']@hotwired\/stimulus["']: ["']\^3.1.0["']/ }
+  FileUtils.touch(package_list)
+  append_file(package_list, "@hotwired/stimulus@^3.1.0\n", verbose: false)
+  say "✅ Enqueued @hotwired/stimulus@^3.1.0 to be added to dependencies"
+end
+
+if !lines.index { |line| line =~ /^\s*["']@hotwired\/stimulus-webpack-helpers["']: ["']\^1.0.1["']/ }
+  FileUtils.touch(package_list)
+  append_file(package_list, "@hotwired/stimulus-webpack-helpers@^1.0.1\n", verbose: false)
+  say "✅ Enqueued @hotwired/stimulus-webpack-helpers@^1.0.1 to be added to dependencies"
+end
+
+if !lines.index { |line| line =~ /^\s*["']webpack-dev-server["']: ["']\^3.11.3["']/ }
+  FileUtils.touch(dev_package_list)
+  append_file(dev_package_list, "webpack-dev-server@^3.11.3\n", verbose: false)
+  say "✅ Enqueued webpack-dev-server@^3.11.3 to be added to dev dependencies"
+end
```

Comment:
Why are you not just running `yarn add [list-of-packages-here]`? Then you can be sure they are installed and that your if checks don't miss because of the version numbers

Reply:
Because each run of yarn 1. takes between 10-30 seconds on my machine and 2. creates a ton of visual output, and having worked with my partner as she learns coding, the more shit a script vomits out, the less likely she is to actually read the output, which is a very bad thing.

It's not just that I'm proud of my solution (I am). I'm also confident that it's an order of magnitude faster for the person running the install. Given that we have packages being installed in multiple steps which are not all guaranteed to run, and there's no way to combine dependencies, devDependencies and removed dependencies in one command, it means we'd have to run yarn 3-7 times during the install and depending on what options they select. One yarn == 🥳